pkg/controller/jobframework/reconciler.go

```diff
@@ -236,6 +252,22 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
+// isParentJobManaged checks whether the parent job is managed by kueue.
+func (r *JobReconciler) isParentJobManaged(ctx context.Context, jobObj client.Object, namespace string) (bool, error) {
+	owner := metav1.GetControllerOf(jobObj)
+	if owner == nil {
+		return false, errChildJobOwnerNotFound
+	}
+	parentJob := KnownWorkloadOwnerObject(owner)
+	if parentJob == nil {
+		return false, fmt.Errorf("workload owner %v is unknown", owner)
```

Comment:
`fmt.Errorf("owner %v: %w", owner, unknownWorkloadOwner)`

This will allow you to use `cmp` directly.

Reply:
That way seems not to allow me to use `cmp`...

https://github.com/kubernetes-sigs/kueue/pull/821/commits/defd1b510ffa72b8a5e8ae40a3ae89c3c2f69ebb

```shell
--- FAIL: TestIsParentJobManaged (0.01s)
    --- FAIL: TestIsParentJobManaged/child_job_has_ownerReference_with_unknown_workload_owner (0.00s)
        reconciler_test.go:112: Unexpected error (-want,+got):
              (*fmt.wrapError)(
            -   &⟪0x01400078e860⟫{
            -           msg: "workload owner &OwnerReference{Kind:Job,Name:test-job-parent,UID:test-job-parent,APIVersion:batch/v1,Controller:*true,BlockOwnerDeletion:nil,}: workload owner is unknown",
            -           err: &⟪0x0140005279a0⟫"errors".errorString{s: "workload owner is unknown"},
            -   },
            +   &⟪0x014000295a00⟫{
            +           msg: "workload owner &OwnerReference{Kind:Job,Name:test-job-parent,UID:test-job-parent,APIVersion:batch/v1,Controller:*true,BlockOwnerDeletion:nil,}: workload owner is unknown",
            +           err: &⟪0x0140005279a0⟫"errors".errorString{s: "workload owner is unknown"},
            +   },
              )
    --- FAIL: TestIsParentJobManaged/child_job_has_ownerReference_with_known_non-existing_workload_owner (0.00s)
        reconciler_test.go:112: Unexpected error (-want,+got):
              (*fmt.wrapError)(
            -   &⟪0x01400078e880⟫{
            -           msg: `workload owner not found: mpijobs.kubeflow.org "test-job-parent" not found`,
            -           err: &⟪0x014000782c80⟫"k8s.io/apimachinery/pkg/api/errors".StatusError{
            -                   ErrStatus: "k8s.io/apimachinery/pkg/apis/meta/v1".Status{
            -                           Status:  "Failure",
            -                           Message: `mpijobs.kubeflow.org "test-job-parent" not found`,
            -                           Reason:  "NotFound",
            -                           Details: &⟪0x01400060cc60⟫"k8s.io/apimachinery/pkg/apis/meta/v1".StatusDetails{Name: "test-job-parent", Group: "kubeflow.org", Kind: "mpijobs"},
            -                           Code:    404,
            -                   },
            -           },
            -   },
            +   &⟪0x01400078fc00⟫{
            +           msg: `workload owner not found: mpijobs.kubeflow.org "test-job-parent" not found`,
            +           err: &⟪0x014000630640⟫"k8s.io/apimachinery/pkg/api/errors".StatusError{
            +                   ErrStatus: "k8s.io/apimachinery/pkg/apis/meta/v1".Status{
            +                           Status:  "Failure",
            +                           Message: `mpijobs.kubeflow.org "test-job-parent" not found`,
            +                           Reason:  "NotFound",
            +                           Details: &⟪0x01400060ca20⟫"k8s.io/apimachinery/pkg/apis/meta/v1".StatusDetails{Name: "test-job-parent", Group: "kubeflow.org", Kind: "mpijobs"},
            +                           Code:    404,
            +                   },
            +           },
            +   },
              )
FAIL
FAIL    sigs.k8s.io/kueue/pkg/controller/jobframework   0.907s
FAIL
```

Reply:
@alculquicondor As a result, I think we can not use `cmp` since the errors refer to other memory addresses in the `child job has ownerReference with unknown workload owner` and the `child job has ownerReference with known non-existing workload owner` case.

So, I'd like to revert https://github.com/kubernetes-sigs/kueue/commit/defd1b510ffa72b8a5e8ae40a3ae89c3c2f69ebb. Any thought?

Reply:
use the same memory address in `wantError`.

If that doesn't work then, yes, you can revert.